nntrainer/layers/time_dist.cpp

```diff
@@ -23,6 +23,63 @@ namespace nntrainer {
 
 const std::string TimeDistLayer::type = "time_dist";
 
+static void reshape(Tensor &m) {
+  TensorDim d = m.getDim();
+  m.reshape({d[2], d[1], d[0], d[3]});
+}
+
+void TimeDistLayer::setPosition() {
+  positions[0] = net_input[0]->getVariableRef().getData();
+  positions[1] = net_input[0]->getGradientRef().getData();
+  positions[2] = net_hidden[0]->getVariableRef().getData();
+  positions[3] = net_hidden[0]->getGradientRef().getData();
+}
+
+void TimeDistLayer::transposeInOut() {
+
+  Tensor &input_ = net_input[0]->getVariableRef();
+  input_.copy(transposeTensor(input_));
+
+  Tensor &ret_ = net_input[0]->getGradientRef();
+  if (ret_.getData() != positions[0]) {
+    ret_.copy(transposeTensor(ret_));
+  } else {
+    reshape(ret_);
+  }
+
+  Tensor &hval_ = net_hidden[0]->getVariableRef();
+  if (hval_.getData() != positions[0] && hval_.getData() != positions[1]) {
+    hval_.copy(transposeTensor(hval_));
+  } else {
+    reshape(hval_);
+  }
```

Comment:
`setPosition` maps 
position[0] <- input.variable
position[1] <- input.gradient
position[2] <- hidden.variable
position[3] <- hidden.gradient

here it looks index is not as stated above. I wonder if it is intended?

Reply:
1. Start with check input.variable is transposed. ( this is always needed ). 
2. for input.gradient, check if it is same position with input.variable (position[0])
3. hidden.varibale : check if it is the same position with input.variable, input.gradient (position[0], position [1])
4. hidden.gradient : check if it is the same position with position[0], position[1], position[2]
 If it is same, just change shape of tensor, cause it is transposed already.

Reply:
Clarified offline. Thank you for the explanation.  👍